jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -241,90 +259,52 @@ public URI getURI()
-         * a trailing slash if it's a directory, (something
-         * not all FileSystems seem to support)
-         */
-        if (!URIUtil.equalsIgnoreEncodings(uri, toUri(path)))
+        if (!aliasResolved)
         {
-            try
+            aliasResolved = true;
+            canonicalPath = resolveCanonicalPath();
+            if (canonicalPath == null)
             {
-                // Use normalized path to get past navigational references like "/bar/../foo/test.txt"
-                Path ref = Paths.get(uri.normalize());
-                return ref.toRealPath();
+                alias = true;
             }
-            catch (IOException ioe)
+            else
             {
-                // If the toRealPath() call fails, then let
-                // the alias checking routines continue on
-                // to other techniques.
-                LOG.trace("IGNORED", ioe);
+                /* If the path and canonicalPath are the same also check
+                 * the Path class has already normalized in the constructor
+                 * from the URI e.g. input path "aa./foo.txt"
+                 * from an #resolve(String) is normalized away during
+                 * the creation of a Path object reference.
+                 * If the URI is different from the Path.toUri() then
+                 * we will just use the original URI to construct the
+                 * alias reference Path.
+                 *
+                 * // On Windows
+                 *  PathResource resource         = PathResource("C:/temp");
+                 *  PathResource child            = resource.resolve("aa./foo.txt");
+                 *  child.exists()                == true
+                 *  child.isAlias()               == true
+                 *  child.toUri()                 == "file:///C:/temp/aa./foo.txt"
+                 *  child.getPath().toUri()       == "file:///C:/temp/aa/foo.txt"
+                 *  child.getCanonicalURI()       == "file:///C:/temp/aa/foo.txt"
+                 */
+                alias = !isSameName(path, canonicalPath) || !URIUtil.equalsIgnoreEncodings(uri, toUri(path));
             }
         }
+    }
 
-        if (!abs.isAbsolute())
-            abs = path.toAbsolutePath();
-
-        // Any normalization difference means it's an alias,
-        // and we don't want to bother further to follow
-        // symlinks as it's an alias anyway.
-        Path normal = path.normalize();
-        if (!isSameName(abs, normal))
-            return normal;
-
+    private Path resolveCanonicalPath()
+    {
         try
         {
-            if (Files.isSymbolicLink(path))
-                return path.getParent().resolve(Files.readSymbolicLink(path));
-            if (Files.exists(path))
-            {
-                Path real = abs.toRealPath();
-                if (!isSameName(abs, real))
-                    return real;
-            }
+            return path.normalize().toRealPath();
```

Comment:
This usage of `Path.toRealPath()` (with no arguments) is returning the FileSystem real name.

If you want the in-API version (not the true FileSystem name) then you have to pass in the NO_FOLLOW_LINKS argument.

Since you are dead set on using `canonicalPath` as the name of the field, at least use the API properly to actually get the old-school in-API canonicalization techniques out of this API.

I *really* don't like this.

Reply:
I have reverted the name back to `getTargetURI` for now.